tests/cupyx_tests/fallback_mode_tests/test_fallback.py

```diff
@@ -147,3 +149,133 @@ def test_func_not_in_numpy(self):
+    @numpy_fallback_equal()
+    def test_ndarray_tobytes(self, xp):
+
+        a = xp.array([1, 2, 3])
+
+        return a.tobytes()
+
+    @numpy_fallback_equal()
+    def test_ndarray_min(self, xp):
+
+        a = xp.array([1, 2, 0, 4])
+
+        return a.min()
+
+    @numpy_fallback_equal()
+    def test_ndarray_argmin(self, xp):
+
+        a = xp.array([[1, 2, 3], [7, 8, 9]])
+
+        return a.argmin()
+
+    @numpy_fallback_equal()
+    def test_ndarray_argmin_kwargs(self, xp):
+
+        a = xp.array([[1, 2, 3], [7, 8, 9]])
+
+        return a.argmin(axis=0)
+
+    def test_magic_methods(self):
+
+        a = fallback_mode.numpy.array([1, 2, 3])
+        b = fallback_mode.numpy.array([9, 8, 7])
+
+        # __add__
+        x = a + b
+        res = cupy.array([10, 10, 10])
+        assert isinstance(x, utils.ndarray)
+        testing.assert_array_equal(x._array, res)
+
+        # __iadd__
+        x += x
+        res += res
+        testing.assert_array_equal(x._array, res)
+
+        # __mul__ with integer
+        x = x * 5
+        res = res * 5
+        testing.assert_array_equal(x._array, res)
+
+        # __str__
+        assert str(x) == str(res)
+
+        # __neg__ single arg: self
+        testing.assert_array_equal((-x)._array, -res)
+
+        # __len__
+        assert len(x) == len(res)
+
+        # conversion __int__, __float__
+        assert int(fallback_mode.numpy.array([3])) == 3
+        assert float(fallback_mode.numpy.array([3])) == 3.0
+
+    def test_getitem(self):
+
+        x = fallback_mode.numpy.array([1, 2, 3])
+
+        # single element
+        assert int(x[2]) == 3
+
+        # slicing
+        res = cupy.array([1, 2, 3])
+        testing.assert_array_equal(x[:2]._array, res[:2])
+
+    def test_setitem(self):
+
+        x = fallback_mode.numpy.array([1, 2, 3])
+
+        # single element
+        x[2] = 99
+        res = cupy.array([1, 2, 99])
+        testing.assert_array_equal(x._array, res)
+
+        # slicing
+        y = fallback_mode.numpy.array([11, 22])
+        x[:2] = y
+        res = cupy.array([11, 22, 99])
+        testing.assert_array_equal(x._array, res)
+
+    @numpy_fallback_equal()
+    def test_ndarray_shape(self, xp):
+
+        x = xp.arange(20)
+        x = x.reshape(4, 5)
+
+        return x.shape
+
+    def test_ndarray_shape_creation(self):
+
+        a = fallback_mode.numpy.ndarray((4, 5))
+        b = cupy.ndarray((4, 5))
+        assert a.shape == (4, 5)
+        testing.assert_array_almost_equal(a._array, b, decimal=5)
+
+    def test_instancecheck_ndarray(self):
+
+        a = fallback_mode.numpy.array([1, 2, 3])
+        assert isinstance(a, fallback_mode.numpy.ndarray)
+
+        b = fallback_mode.numpy.ndarray((2, 3))
+        assert isinstance(b, fallback_mode.numpy.ndarray)
+
+    def test_instancecheck_dtypes(self):
+
+        a = fallback_mode.numpy.int32(3)
+        b = fallback_mode.numpy.int64(6)
+
+        assert isinstance(1, fallback_mode.numpy.int)
```

Comment:
```
>>> numpy.int is cupy.int
True
```
`cupy.int` just mirrors `numpy.int` How about returning `numpy_object` if `numpy_object is cupy_object` is `True` in `_RecursiveAttr.__getitem__`?

Reply:
This behavior is only with some `ScalarTypes` only. And you mean `_RecursiveAttr.__getattr__`. Right?

Reply:
Could you fix the following error?
```
>>> from cupyx.fallback_mode import numpy
>>> numpy.array([1, 2, 3], dtype=numpy.int64)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/imanishi/cupy/cupyx/fallback_mode/fallback.py", line 87, in __call__
    return utils._call_cupy(self._cupy_object, args, kwargs)
  File "/home/imanishi/cupy/cupyx/fallback_mode/utils.py", line 28, in _call_cupy
    res = func(*args, **kwargs)
  File "/home/imanishi/cupy/cupy/creation/from_data.py", line 41, in array
    return core.array(obj, dtype, copy, order, subok, ndmin)
  File "cupy/core/core.pyx", line 1714, in cupy.core.core.array
    cpdef ndarray array(obj, dtype=None, bint copy=True, order='K',
  File "cupy/core/core.pyx", line 1791, in cupy.core.core.array
    a = _send_object_to_gpu(obj, dtype, order, ndmin)
  File "cupy/core/core.pyx", line 1860, in cupy.core.core._send_object_to_gpu
    a_cpu = numpy.array(obj, dtype=dtype, copy=False, order=order,
TypeError: data type not understood
```

Reply:
Just debugged it. It's happening because of following behavior of numpy:
```
isinstance(np.int64, np.ScalarType) # False
isinstance(np.int64(5), np.ScalarType) # True
```
So, when we do `dtype=numpy.int64`, `_RecursiveAttr` class instance is passed.

Reply:
Yes. `fallback_mode.numpy.int64` is `_RecursiveAttr` type value, which has same object in `self._numpy_object` and `self._cupy_object`. How about returning this object directly?

Reply:
Yeah, I'm thinking the same. We should return it as we did with scalar types. But how can we check for it? I'm searching for way like `isinstance(numpy.int64, numpy.dtype) # True`, but could not find any. Do you know if there exists something like this?

Reply:
> return numpy_object if numpy_object is cupy_object is True
Sorry, I get it now

Reply:
Done